cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -136,6 +137,9 @@ type ClusterStateRegistry struct {
 	cloudProviderNodeInstancesCache    *utils.CloudProviderNodeInstancesCache
 	interrupt                          chan struct{}
 	maxNodeProvisionTimeProvider       maxNodeProvisionTimeProvider
+	// scaleDownDelayStatus is for tracking nodegroups which should be
+	// in scaledown cooldown
+	scaleDownDelayStatus *node_processors.ScaleDownDelayStatus
```

Comment:
I have moved `scaleDownDelayStatus` from `static_autoscaler` to clusterstate because adding it here makes it easier to update `scaleDownDelayStatus`  after scale-up [here](https://github.com/kubernetes/autoscaler/pull/5729/files#diff-95d98f54ac57fc3c60c23c1daca659d1fb752fc1822d32e42e12614b96331183R594). 

I might move this again after I give a thought to https://github.com/kubernetes/autoscaler/pull/5729#discussion_r1186092461

Reply:
I think `scaleDownDelayStatus` can use a better name. Any suggestions around the name are welcome :pray:

Reply:
P.S.: I might be able to use [`csr.scaleUpRequests`](https://github.com/kubernetes/autoscaler/blob/e90529bae270d61f345bc01b71722295d5a469ac/cluster-autoscaler/clusterstate/clusterstate.go#L224) instead of a separate struct for tracking scale up times for `ScaleDownDelayAfterAdd`.

Reply:
On the flip side, we might still need it for `--scale-down-delay-after-delete` and `--scale-down-delay-after-failure`. Having a separate struct might not be such a bad idea (at the cost of duplicating some logic from `csr.scaleUpRequests`). Any feedback around this is welcome :pray:

Reply:
Seems like I might be able to use [`csr.scaleUpFailures`](https://github.com/kubernetes/autoscaler/blob/e90529bae270d61f345bc01b71722295d5a469ac/cluster-autoscaler/clusterstate/clusterstate.go#L144-L146) for  `--scale-down-delay-after-failure` and [`csr.scaleDownRequests`](https://github.com/kubernetes/autoscaler/blob/e90529bae270d61f345bc01b71722295d5a469ac/cluster-autoscaler/clusterstate/clusterstate.go#L256-L261) for `--scale-down-delay-after-delete`

Reply:
Need a little more research around this idea to see if I can actually use the existing fields on `clusterstate`